packages/vdom-extension/src/index.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { JupyterFrontEnd, ILayoutRestorer } from '@jupyterlab/application';
+
+import { InstanceTracker } from '@jupyterlab/apputils';
+
+import { MimeDocumentFactory, MimeDocument } from '@jupyterlab/docregistry';
+
+import { INotebookTracker } from '@jupyterlab/notebook';
+
+import { IRenderMimeRegistry } from '@jupyterlab/rendermime';
+
+import { RenderedVDOM } from '@jupyterlab/vdom';
+
+/**
+ * The CSS class for a VDOM icon.
+ */
+const CSS_ICON_CLASS = 'jp-MaterialIcon jp-VDOMIcon';
+
+/**
+ * The MIME type for VDOM.
+ */
+export const MIME_TYPE = 'application/vdom.v1+json';
+
+/**
+ * The name of the factory that creates VDOM widgets.
+ */
+const FACTORY_NAME = 'VDOM';
+
+const plugin = {
+  id: '@jupyterlab/vdom-extension:factory',
+  requires: [IRenderMimeRegistry, INotebookTracker, ILayoutRestorer],
+  autoStart: true,
+  activate: (
+    app: JupyterFrontEnd,
+    rendermime: IRenderMimeRegistry,
+    notebooks: INotebookTracker,
+    restorer: ILayoutRestorer
+  ) => {
+    const tracker = new InstanceTracker<MimeDocument>({
```

Comment:
@ian-r-rose I observed that other non-extension packages (e.g. htmlviewer) export a tracker instance. Should this vdom package export the tracker instance or can vdom-extension create one itself?

Reply:
Usually the base package defines the tracker token, and then the extension `provide`s it. This is so that other extensions can import the token without including the `*-extension` package.

Reply:
It's not required, however. If you want to do it here, that's cool, but I think it can also be done in a follow-up.

Reply:
Can you take a look at https://github.com/jupyterlab/jupyterlab/pull/5670/commits/6d89b1cc52b3be7a75a761649fe14a8a33e7925f. I used htmlviewer and htmlviewer-extension as reference.